### PR TITLE
Refactor and test company aggregation for tms sectors

### PR DESCRIPTION
**Pull request by jacobvjk**
relates to #46 

This PR:
- refactors `calculate_company_aggregate_alignment_tms()` and splits some functionality out into sub functions
- adds unit tests

Note:
- the remaining functions will get refactored and unit tested in separate PRs to keep the PR size in check

### CODE REVIEW

File: R/calculate_company_alignment_metric.R [hunk 1]
```diff
@@ -241,36 +241,13 @@ calculate_company_aggregate_alignment_tms <- function(data,
...
-      c(
-        "group_id", "name_abcd", "sector", "activity_unit", "region",
-        "scenario_source", "scenario", "year", "direction", "total_deviation",
-        "technology_share_by_direction", "alignment_metric"
-      )
+    add_net_absolute_scenario_value(target_scenario = target_scenario) %>%
+    add_total_deviation() %>%
+    calculate_company_alignment_metric(scenario = scenario)  %>%
+    dplyr::arrange(
+      .data$group_id, .data$sector, .data$name_abcd, .data$region, .data$year
```
Comment (cjyetman) [inline R/calculate_company_alignment_metric.R:248]: I know @AlexAxthelm strongly prefers lists of arguments like this to be one per line

**Review by cjyetman — APPROVED** [review]

File: R/calculate_company_alignment_metric.R (hunk 1 above)
Comment (jacobvjk) [reply R/calculate_company_alignment_metric.R:248]: do you mean this?
```
dplyr::arrange(
   .data$group_id,
   .data$sector,
   .data$name_abcd,
   .data$region,
   .data$year
)
```

Comment (cjyetman) [reply R/calculate_company_alignment_metric.R:248]: yup

Comment (jacobvjk) [reply R/calculate_company_alignment_metric.R]: changed

**Pull request merged**